src/javascript/app_2/Stores/base_store.js

```diff
@@ -62,6 +64,11 @@ export default class BaseStore {
             enumerable: false,
             writable  : true,
         });
+        Object.defineProperty(this, 'store_name', {
+            value     : store_name,
```

Comment:
@OskarAhl I'd be better to throw an exception here when the `sotre_name` doesn't have a valid value since the `null` or invalid value affects the functionality of the `BaseStore` to load and save the snapshot to the storage. So it's better to prevent developers to pass null when they want to use the autosave feature of the base_store.

e.g. you can throw an exception when the `store_name` is null or invalid but one or both of `local_storage_properties` and 'session_storage_properties` have value.